### PR TITLE
Add ability to create json files from issues

### PR DESCRIPTION
**Pull request by feerrenrut**
See #8 

>The planned procedure to submit add-ons to the addon-datastore repo has security issues, as shown in: https://github.com/nvaccess/addon-datastore/pull/35
>This intends to provide a way to add and validate add-on metadata by creating issues in addon-datastore, which maybe automerged if checks pass.

### CODE REVIEW

File: _validate/createJson.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+import tempfile
+import zipfile
+from dataclasses import dataclass
+import os
+import sys
+sys.path.append(os.path.dirname(__file__))
+# E402 module level import not at top of file
+import sha256  # noqa:E402
+from addonManifest import AddonManifest  # noqa:E402
+del sys.path[-1]
```
Comment (seanbudd) [inline _validate/createJson.py]: what are the sys.path lines for?

Comment (seanbudd) [reply _validate/createJson.py]: ah I see from the other file - can you copy across those comments

File: _tests/test_createJson.py [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+CREATED_SUBMISSION_JSON_FILE = os.path.join(OUTPUT_DIR, ADDON_ID, ADDON_FILENAME)
+
+
+def getValidAddonSubmission() -> validate.JsonObjT:
+	with open(VALID_JSON) as f:
+		submission = json.load(f)
+	return submission
+
+
+def getAddonManifest():
```
Comment (seanbudd) [inline _tests/test_createJson.py:34]:
```suggestion
def getAddonManifest() -> addonManifest.AddonManifest:
```

**Comment by feerrenrut** [comment]
This probably should still be a draft. I'm actively working on this.

File: _tests/test_createJson.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+
+
+class IntegrationTestCreateJson(unittest.TestCase):
+	""" Integration tests.
+	- The JSON file is created (written to the filesystem).
+	- The output is then loaded and checked for correctness.
+	"""
+	def setUp(self):
+		self.outputDir = os.path.join(OUTPUT_DATA_PATH, "createJsonOutput")
+		self.maxDiff = None
```
Comment (nvdaes) [inline _tests/test_createJson.py]:
```suggestion
```

**Review by nvdaes — CHANGES_REQUESTED** [review]
I've seen this to learn looking at changes, since I contributed this. Seems that the self.maxDiff is not used and can be removed.
About the license, I thought that add-ons must be licensed undr GPLv2, like NVDA. If this is not the case, this maybe documentet to know what licenses can be used.

File: _tests/test_createJson.py (hunk 3 above)
Comment (feerrenrut) [reply _tests/test_createJson.py]: This is used by `unittest.Testcase` if not set you only get a small sample of the diff for the dictionaries.

**Comment by feerrenrut** [comment]
There is a [section specifically about plugins and drivers](https://github.com/nvaccess/nvda/blob/master/copying.txt#L292)
I'm not confident to say whether or not other licenses may be be used.
If we are going to enforce a value, that should happen on the usage side of this workflow, so that there is only one place that needs to be updated if the approach needs to change.

I'd prefer not to enforce a license value, it would hide the concern from the author / submitter. They must be actively consider what license they are distributing under and ensure the license is valid / meets requirements etc.

**Comment by nvdaes** [comment]
Thanks for explanations. I"ll follow related PR to create issues in addon-datastore repo.

Enviado desde mi iPhone

**Comment by seanbudd** [comment]
A reference to https://github.com/nvaccess/Internal-issue-tracking/issues/449 is missing from this PR, and vice versa

File: _validate/createJson.py [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+	)
+
+	filePath = buildOutputFilePath(data, parentDir)
+
+	with open(filePath, "wt") as f:
+		json.dump(data, f, indent="\t")
+	print(f"Wrote json file: {filePath}")
+
+
+def buildOutputFilePath(data, parentDir):
```
Comment (seanbudd) [inline _validate/createJson.py]:
```suggestion
def buildOutputFilePath(data, parentDir) -> os.PathLike:
```

File: _validate/createJson.py [hunk 5]
```diff
@@ -0,0 +1,179 @@
...
+def _createDictMatchingJsonSchema(
+		manifest: AddonManifest,
+		sha: str,
+		channel: str,
+		publisher: str,
+		sourceUrl: str,
+		url: str,
+		licenseName: str,
+		licenseUrl: str,
+):
```
Comment (seanbudd) [inline _validate/createJson.py]:
```suggestion
) -> Dict[str, str]:
```

File: _tests/test_createJson.py (hunk 3 above)
Comment (seanbudd) [reply _tests/test_createJson.py]: This might be worth adding a comment to explain

**Comment by feerrenrut** [comment]
Given review actions were minor and have all been addressed, I'm going to merge this, so I can progress https://github.com/nvaccess/addon-datastore/pull/37

**Pull request merged**